src/consensus/pbft/libbyz/prepare.cpp

```diff
@@ -27,6 +28,11 @@ Prepare::Prepare(View v, Seqno s, Digest& d, Principal* dst, bool is_signed) :
   rep().seqno = s;
   rep().digest = d;
 
+  if (id < 0)
+  {
+    rep().id = pbft::GlobalState::get_node().id();
+  }
+
```

Comment:
In what circumstance can this happened? can you please add a comment?

Reply:
This is to indicate that the replica id is not set (see above for when it will/should be set), so that means that the the replica that is creating the Prepare is the same replica issuing the Prepare, so we can just set the node id